packages/fast-usdc/proposals/start-fast-usdc.js

```diff
@@ -0,0 +1,163 @@
+
+  const marshaller = await E(board).getPublishingMarshaller();
+  return { storageNode, marshaller };
+};
+
+/**
+ * @typedef {{ watcherAddress: string, feeAccountAddress: string }} QuickSendConfig
+ */
+
+/**
+ * @param {BootstrapPowers & {
+ *   installation: PromiseSpaceOf<{
+ *     fastUsdc: Installation<FastUsdcSF>;
+ *   }>;
+ *   instance: PromiseSpaceOf<{
+ *     fastUsdc: Instance<FastUsdcSF>;
+ *   }>;
+ *   brand: PromiseSpaceOf<{ USDC: Brand<'nat'> }>;
+ *   issuer: PromiseSpaceOf<{ USDC: Issuer<'nat'> }>;
+ * }} powers
+ * @param {{ options?: { fastUsdc?: QuickSendConfig } }} config
+ */
+export const startFastUsdc = async (
+  {
+    consume: {
+      agoricNames,
+      board,
+      chainStorage,
+      chainTimerService: timerService,
+      localchain,
+      cosmosInterchainService,
+      startUpgradable,
+    },
+    installation: {
+      consume: { fastUsdc },
+    },
+    instance: {
+      produce: { fastUsdc: produceInstance },
+    },
+  },
+  config = {},
+) => {
+  trace('startFastUsdc');
+  const { feeAccountAddress = 'agoric1fee' } = config.options?.fastUsdc || {};
+
+  await null;
+  const USDC = {
+    brand: await E(agoricNames).lookup('brand', 'USDC'),
+    issuer: await E(agoricNames).lookup('issuer', 'USDC'),
+  };
+  const terms = harden({
+    poolFee: AmountMath.make(USDC.brand, 100n), // TODO: parameterize
+    contractFee: AmountMath.make(USDC.brand, 30n),
+    feeAccountAddress,
+  });
+  const { storageNode, marshaller } = await makePublishingStorageKit(
+    'fastUsdc',
+    { board, chainStorage },
+  );
+  assert(storageNode);
+
+  const privateArgs = await deeplyFulfilledObject(
+    harden({
+      agoricNames,
+      localchain,
+      orchestrationService: cosmosInterchainService,
+      storageNode,
+      timerService,
+      marshaller,
+    }),
+  );
+
+  /**
+   * @type {StartUpgradableOpts<FastUsdcSF>}
+   */
+  const startOpts = {
+    label: 'fastUsdc',
+    installation: fastUsdc,
+    issuerKeywordRecord: harden({ USDC: USDC.issuer }),
+    terms,
+    privateArgs,
+  };
+
+  const { instance } = await E(startUpgradable)(startOpts);
+
+  produceInstance.reset();
+  produceInstance.resolve(instance);
+  trace('startFastUsdc done', instance);
+};
+harden(startFastUsdc);
+
+/**
+ * @param {{
+ *   restoreRef: (b: ERef<ManifestBundleRef>) => Promise<Installation<FastUsdcSF>>;
+ * }} utils
+ * @param {{
+ *   installKeys: { fastUsdc: ERef<ManifestBundleRef> };
+ *   options?: { fastUsdc?: QuickSendConfig };
+ * }} param1
+ */
+export const getManifestForQuickSend = (
+  { restoreRef },
+  { installKeys, options },
+) => {
+  return {
+    /** @type {BootstrapManifest} */
+    manifest: {
+      [startFastUsdc.name]: {
+        consume: {
+          chainStorage: true,
+          chainTimerService: true,
+          localchain: true,
+          cosmosInterchainService: true,
+
+          // limited distribution during MN2: contract installation
```

Comment:
> // limited distribution during MN2: contract installation

curious - what's meant by this comment?

Reply:
anticipating MN3 in which anyone can install

Reply:
anticipating MN3 in which anyone can install